### ISSUE TITLE
Last commit broke Lemonbar

### ISSUE THREAD

**Issue by vlada-dudr** [krypt-n/bar#36]
It skips pieces of data I send in, so sometimes I see nothing, or only parts of the bar I wanted to see... So in the end I solved by recompiling an older revision.

**Comment by AdamNiederer** [comment]
I'm also experiencing this issue. Would adding an option to set the character for which a re-paint happens be a reasonable compromise between the old and new behavior? It's currently hardcoded and I'm sure a ton of scripts rely on `\0` being ignored and `\n` repainting, as mine did.

**Comment by alexd2580** [comment]
Yep, after investigating different versions and issues I can confirm that the last commit, which is not yet (since 2016) merged into `LemonBoy/bar`, does indeed break generator scripts which use multiple statements to print the status. I myself am using `lemonbar` inside a C++ status generator, and even with buffering (maybe i missed something though) the bug occurs permanently, making `lemonbar` unusable =( And that's right after I found this nice XFT port...

See [this objection](https://github.com/LemonBoy/bar/pull/186#issuecomment-453550042) for a few more details.
